### PR TITLE
Add python 3.12 and run terminator search cv

### PR DESCRIPTION
**Pull request by nzw0301**
## Motivation

https://github.com/optuna/optuna-examples/issues/230 and run a sklearn search cv by the CI.

## Description of the changes

- Add python 3.12
- Run `terminator_search_cv.py`

### CODE REVIEW

File: .github/workflows/terminator.yml [hunk 1]
```diff
@@ -28,11 +28,14 @@ jobs:
...
+        pip install git+https://github.com/optuna/optuna-integration.git
+        python -c 'import optuna_integration'
 
         pip install -r terminator/requirements.txt
     - name: Run examples
       run: |
-        python terminator/terminator_simple.py
         python terminator/terminator_improvement_plot.py
+        python terminator/terminator_search_cv.py
+        python terminator/terminator_simple.py
```
Comment (nzw0301) [inline .github/workflows/terminator.yml:39]: alphabetical order

**Comment by nzw0301** [comment]
I'm not sure why the skelarn's counter example script: https://github.com/optuna/optuna-examples/blob/main/sklearn/sklearn_optuna_search_cv_simple.py, is not executed by the [CI job](https://github.com/optuna/optuna-examples/blob/main/.github/workflows/sklearn.yml). Maybe additional `pandas` dependency?

**Comment by nabenabe0928** [comment]
@nzw0301 We need the dependency for search cv as well.
https://github.com/optuna/optuna-integration/blob/main/optuna_integration/sklearn/sklearn.py

https://github.com/optuna/optuna-integration/blob/main/optuna_integration/sklearn/requirements.txt

**Review by nabenabe0928 — APPROVED** [review]
Thank you for the update, LGTM!

**Pull request merged**